src/dune_pkg/rev_store.ml

```diff
@@ -232,32 +289,48 @@ module Remote = struct
   ;;
 end
 
-let remote_exists { dir } ~name =
-  (* TODO read this directly from .git/config *)
```

Comment:
I'm not convinced that this is a good idea to reimplement git logic and parsing its files in dune. It usually has good machine-readable interfaces.

Reply:
Ok, I've rebased the PR and took out all the locking code, so we can discuss it on the merit of just the `git remote` handling.

The point of removing the `git` invocation is mostly that this is a bit faster and makes us a bit more independent from the `git` CLI (which granted, is rather unlikely to break). The `.git/config` format is fairly stable so just adding the lines in OCaml seems to be a fairly low-risk change and removes 1-2 `git` calls per configured repository.